server/Arcadia.Assistant.Notifications.Push/PushNotificationsDevicesActor.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace Arcadia.Assistant.Notifications.Push
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Akka.Actor;
+    using Akka.Persistence;
+
+    using Arcadia.Assistant.Notifications.Push.Events;
+
+    public class PushNotificationsDevicesActor : UntypedPersistentActor, ILogReceive
+    {
+        private readonly Dictionary<string, HashSet<string>> devicesByEmployeeId = new Dictionary<string, HashSet<string>>();
+
+        public override string PersistenceId => "push-notifications-devices";
+
+        protected override void OnCommand(object message)
+        {
+            switch (message)
+            {
+                case GetDeviceIds msg:
+                    this.GetDeviceIds(msg);
+                    break;
+
+                case RegisterPushNotificationsDevice msg:
+                    this.RegisterDevice(msg);
+                    break;
+
+                case RemovePushNotificationsDevice msg:
+                    this.RemoveDevice(msg);
+                    break;
+
+                default:
+                    this.Unhandled(message);
+                    break;
+            }
+        }
+
+        protected override void OnRecover(object message)
+        {
+            switch (message)
+            {
+                case EmployeeDeviceRegistered ev:
+                    this.OnEmployeeDeviceRegistered(ev);
+                    break;
+
+                case EmployeeDeviceRemoved ev:
+                    this.OnEmployeeDeviceRemoved(ev);
+                    break;
+            }
+        }
+
+        private void GetDeviceIds(GetDeviceIds message)
+        {
+            this.devicesByEmployeeId.TryGetValue(message.EmployeeId, out var deviceIds);
+            this.Sender.Tell(new GetDeviceIds.Success(deviceIds ?? Enumerable.Empty<string>()));
```

Comment:
better to send a copy of deviceIds for thread-safety reasons

Reply:
Done.